core/src/test/scala/unit/kafka/server/ShareFetchAcknowledgeRequestTest.scala

```diff
@@ -113,9 +111,8 @@ class ShareFetchAcknowledgeRequestTest(cluster: ClusterInstance) extends GroupCo
     // Send the share fetch request to the non-replica and verify the error code
     val shareFetchRequest = createShareFetchRequest(groupId, metadata, MAX_PARTITION_BYTES, send, Seq.empty, Map.empty)
     val shareFetchResponse = connectAndReceive[ShareFetchResponse](shareFetchRequest, nonReplicaId)
-    val partitionData = shareFetchResponse.responseData(topicNames).get(topicIdPartition)
-    assertEquals(Errors.NOT_LEADER_OR_FOLLOWER.code, partitionData.errorCode)
-    assertEquals(leader, partitionData.currentLeader().leaderId())
+    // Top level error thrown while fetching the "LATEST" offset for the partition during share partition initialization
+    assertEquals(Errors.NOT_LEADER_OR_FOLLOWER.code, shareFetchResponse.data().errorCode())
```

Comment:
Why do we want to have this behaviour? Shouldn't the exception be at per topic partition level?

Reply:
This is not a top-level error code. It's partition-specific.